src/views/Home/index.js

```diff
@@ -0,0 +1,102 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { Link } from 'react-router-dom';
+
+// Components
+import { GIFComponent, Header } from '../../components';
+
+// Connect
+import Connect from '../../util/connect';
+
+// Styles
+import {
+  HomeContainer,
+  Content,
+  ButtonAddFavorite,
+  ButtonRemoveFavorite,
+  FavoriteContainer
+} from './HomeStyles';
+
+class Home extends Component {
+  constructor() {
+    super();
+    this.searchFunction = this.searchFunction.bind(this);
+    this.handleFavoriteChange = this.handleFavoriteChange.bind(this);
+    this.renderHeart = this.renderHeart.bind(this);
+  }
+
+  renderHeart(id) {
+    const { appState: { favoriteGifs } } = this.props;
```

Comment:
you have kinda the same logic in `GifView`. You could create a component that contains this behavior